src/community/params-extractor/src/main/java/org/geoserver/params/extractor/Filter.java

```diff
@@ -33,34 +47,52 @@ public int getPriority() {
     }
 
     @Override
-    public void init(FilterConfig filterConfig) throws ServletException {}
+    public void init(FilterConfig filterConfig) throws ServletException {
+        GeoServerDataDirectory dataDirectory =
+                GeoServerExtensions.bean(GeoServerDataDirectory.class);
+        if (dataDirectory != null) {
+            USE_AS_SERVLET_FILTER = true;
+            Resource resource = dataDirectory.get(RulesDao.getRulesPath());
```

Comment:
This code duplicates the constructor receiving a data dir, can it be shared in a method that both call?